wgpu-native/src/command/mod.rs

```diff
@@ -249,20 +249,42 @@ pub fn command_encoder_begin_render_pass<B: GfxBackend>(
             "Attachment sample_count must be supported by physical device limits"
         );
         for at in color_attachments.iter() {
-            let sample_count_check = view_guard[at.attachment].samples;
+            let color_attachment_view = &view_guard[at.attachment];
             assert_eq!(
-                sample_count_check, sample_count,
+                color_attachment_view.samples, sample_count,
                 "All attachments must have the same sample_count"
             );
 
-            if let Some(resolve) = unsafe { at.resolve_target.as_ref() } {
+            let color_attachment_texture = &texture_guard[color_attachment_view.texture_id.value];
```

Comment:
Since we access the texture a little bit later anyway, let's move the check there in order to avoid fetching the texture data storage more than necessary. Doing the checks as early as possible is a worthy goal, but only when it doesn't harm our performance.